sdk/identity/Azure.Identity/tests/TenantIdResolverTests.cs

```diff
@@ -61,7 +61,7 @@ public static IEnumerable<object[]> ResolveInputs()
                         disableMultiTenantAuth.Value.ToString());
                 }
 
-                result = TenantIdResolver.Resolve(tenantId, context);
+                result = TenantIdResolver.Resolve(tenantId, context, TenantIdResolver.AllTenants);
```

Comment:
Do we need a test for the new `AuthenticationFailedException` scenario?

Reply:
Added